### PR TITLE
In Log messages (file_name:line_number) links to source code in Android Studio

### PR DESCRIPTION
**Pull request by biafra23**
Change logging of class name and line number to filename and line number. This way Android Studio can link to the source code.

Before:
before

After:
after

### CODE REVIEW

**Comment by stefanhoth** [comment]
:boom: awesome!
On May 21, 2015 7:27 PM, "Dirk Jäckel" <notifications@github.com> wrote:

> Change logging of class name and line number to filename and line number.
> This way Android Studio can link to the source code.
>
> Before:
> [image: before]
> <https://cloud.githubusercontent.com/assets/246473/7754749/445a3646-ffef-11e4-80b1-6ec4588f9114.png>
>
> After:
> [image: after]
> <https://cloud.githubusercontent.com/assets/246473/7754720/26029da0-ffef-11e4-97d8-4cf40936222f.png>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/novoda/notils/pull/60
> Commit Summary
>
>    - Change logging of class name and line number to filename and line
>    number. This way Android Studio can link to the source
>
> File Changes
>
>    - *M* src/main/java/com/novoda/notils/logger/simple/Log.java
>    <https://github.com/novoda/notils/pull/60/files#diff-0> (4)
>
> Patch Links:
>
>    - https://github.com/novoda/notils/pull/60.patch
>    - https://github.com/novoda/notils/pull/60.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/notils/pull/60>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

**Comment by amlcurran** [comment]
No way! that's amazing :100:

**Comment by mr-archano** [comment]
:100:

**Comment by blundell** [comment]
:100: 

but before your screenie says `DownloadService` and after `DownloadProvider`.  I hope this is just a screenshot error.

File: src/main/java/com/novoda/notils/logger/simple/Log.java [hunk 1]
```diff
@@ -177,8 +177,8 @@ private static String getDetailedLog(String msg, int depth) {
         Thread current = Thread.currentThread();
         final StackTraceElement trace = current.getStackTrace()[depth];
         final String filename = trace.getFileName();
-        return "[" + current.getName() + "][" + filename.substring(0, filename.length() - DOT_CLASS) + "."
-                + trace.getMethodName() + ":" + trace.getLineNumber() + "] " + msg;
+        return "[" + current.getName() + "][" + "(" + filename.substring(0, filename.length() - DOT_CLASS) + ".java:"
+                + trace.getLineNumber() + ")." + trace.getMethodName() + "] " + msg;
```
Comment (blundell) [inline src/main/java/com/novoda/notils/logger/simple/Log.java]: why move the line number and method name around? or is this involved in the linking

**Comment by stefanhoth** [comment]
@Blundell DownloadPotato, DownloadPotato ;-)
On May 21, 2015 11:39 PM, "Paul Blundell" <notifications@github.com> wrote:

> [image: :100:]
>
> but before your screenie says DownloadService and after DownloadProvider.
> I hope this is just a screenshot error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/notils/pull/60#issuecomment-104427955>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

File: src/main/java/com/novoda/notils/logger/simple/Log.java (hunk 1 above)
Comment (stefanhoth) [inline src/main/java/com/novoda/notils/logger/simple/Log.java]: As I understand it the linking happens automatically if Name and line
number follow each other. So the method name was destroying that.
On May 21, 2015 11:40 PM, "Paul Blundell" <notifications@github.com> wrote:

> In src/main/java/com/novoda/notils/logger/simple/Log.java
> <https://github.com/novoda/notils/pull/60#discussion_r30850719>:
>
> > @@ -177,8 +177,8 @@ private static String getDetailedLog(String msg, int depth) {
> >          Thread current = Thread.currentThread();
> >          final StackTraceElement trace = current.getStackTrace()[depth];
> >          final String filename = trace.getFileName();
> > -        return "[" + current.getName() + "][" + filename.substring(0, filename.length() - DOT_CLASS) + "."
> > -                + trace.getMethodName() + ":" + trace.getLineNumber() + "] " + msg;
> > +        return "[" + current.getName() + "][" + "(" + filename.substring(0, filename.length() - DOT_CLASS) + ".java:"
> > +                + trace.getLineNumber() + ")." + trace.getMethodName() + "] " + msg;
>
> why move the line number and method name around? or is this involved in
> the linking
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/notils/pull/60/files#r30850719>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

Comment (stefanhoth) [inline src/main/java/com/novoda/notils/logger/simple/Log.java]: Also wouldn't this be nicer to read with a `StringBuilder`? 

Boy scout, maybe?

Comment (amlcurran) [inline src/main/java/com/novoda/notils/logger/simple/Log.java]: Or string.format

**Comment by biafra23** [comment]
I did some boyscouting.

**Comment by biafra23** [comment]
@blundell the screenshots are just examples to show the linkyfication. I did not screenshot the same log output.

**Comment by danybony** [comment]
Great work @biafra23 :100:

**Comment by xrigau** [comment]
epic :+1:

**Pull request merged**